hadoop-hdds/server-scm/src/main/java/org/apache/hadoop/hdds/scm/server/SCMBlockProtocolServer.java

```diff
@@ -261,26 +264,33 @@ public List<DeleteBlockGroupResult> deleteKeyBlocks(
     List<DeleteBlockGroupResult> results = new ArrayList<>();
     Map<String, String> auditMap = Maps.newHashMap();
     ScmBlockLocationProtocolProtos.DeleteScmBlockResult.Result resultCode;
+    perfMetrics = scm.getPerfMetrics();
     Exception e = null;
+    long startNanos = Time.monotonicNowNanos();
     try {
       scm.getScmBlockManager().deleteBlocks(keyBlocksInfoList);
+      perfMetrics.updateDeleteKeySuccessStats(startNanos);
       resultCode = ScmBlockLocationProtocolProtos.
           DeleteScmBlockResult.Result.success;
     } catch (IOException ioe) {
       e = ioe;
+      perfMetrics.updateDeleteKeyFailureStats(startNanos);
       LOG.warn("Fail to delete {} keys", keyBlocksInfoList.size(), ioe);
       switch (ioe instanceof SCMException ? ((SCMException) ioe).getResult() :
           IO_EXCEPTION) {
       case SAFE_MODE_EXCEPTION:
+        perfMetrics.updateDeleteKeyFailureStats(startNanos);
```

Comment:
Updating the `deleteKeyFailureStats` once at the beginning of the catch block (`L277`) should suffice.
```suggestion
```